File: examples/src/WinForms/GetActiveConnectionIPAddress/CS/GetActiveConnectionIPAddress/MainForm.cs
using System;
using System.Windows.Forms;

namespace DotRas.Samples.GetActiveConnectionIPAddress
{
    public partial class MainForm : Form
    {
        public MainForm()
        {
            InitializeComponent();
        }

        private void LoadConnectionsComboBox()
        {
            ConnectionsComboBox.Items.Add(new ComboBoxItem("Choose one...", null));

            foreach (var connection in RasConnection.GetActiveConnections())
            {
                ConnectionsComboBox.Items.Add(new ComboBoxItem(connection.EntryName, connection.EntryId));
            }

            ConnectionsComboBox.SelectedIndex = 0;
        }

        private void MainForm_Load(object sender, EventArgs e)
        {
            LoadConnectionsComboBox();
        }

        private void GetAddressButton_Click(object sender, EventArgs e)
        {
            foreach (var connection in RasConnection.GetActiveConnections())
            {
                if (connection.EntryId != (Guid) ((ComboBoxItem) ConnectionsComboBox.SelectedItem).Value) continue;
                var ipAddresses = (RasIPInfo)connection.GetProjectionInfo(RasProjectionType.IP);
                if (ipAddresses == null) continue;

                ClientAddressTextBox.Text = ipAddresses.IPAddress.ToString();
                ServerAddressTextBox.Text = ipAddresses.ServerIPAddress.ToString();
            }
        }

        private void ConnectionsComboBox_SelectedIndexChanged(object sender, EventArgs e)
        {
            GetAddressButton.Enabled = ConnectionsComboBox.SelectedIndex > 0;
        }
    }
}

File: examples/src/WinForms/GetActiveConnectionIPAddress/VB/GetActiveConnectionIPAddress/MainForm.vb
Imports DotRas

Public Class MainForm

    Private Sub LoadConnectionsComboBox()
        Me.ConnectionsComboBox.Items.Add(New ComboBoxItem("Choose one...", Nothing))

        For Each connection As RasConnection In RasConnection.GetActiveConnections()
            Me.ConnectionsComboBox.Items.Add(New ComboBoxItem(connection.EntryName, connection.EntryId))
        Next

        Me.ConnectionsComboBox.SelectedIndex = 0
    End Sub

    Private Sub MainForm_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        Me.LoadConnectionsComboBox()
    End Sub

    Private Sub GetAddressButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles GetAddressButton.Click
        For Each connection As RasConnection In RasConnection.GetActiveConnections()
            If connection.EntryId = CType(CType(Me.ConnectionsComboBox.SelectedItem, ComboBoxItem).Value, Guid) Then
                Dim ipAddress As RasIPInfo = CType(connection.GetProjectionInfo(RasProjectionType.IP), RasIPInfo)

                If ipAddress IsNot Nothing Then
                    Me.ClientAddressTextBox.Text = ipAddress.IPAddress.ToString()
                    Me.ServerAddressTextBox.Text = ipAddress.ServerIPAddress.ToString()
                End If
            End If
        Next
    End Sub

    Private Sub ConnectionsComboBox_SelectedIndexChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles ConnectionsComboBox.SelectedIndexChanged
        Me.GetAddressButton.Enabled = Me.ConnectionsComboBox.SelectedIndex > 0
    End Sub

End Class